### PR TITLE
doc(Xamarin): Add privacy info page for Xamarin

### PR DESCRIPTION
**Pull request by lucas-zimerman**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Apple now requires the user projects to include a Privacy manifest file on their project, this PR aids the user on how to add this information on an Xamarin project and also informs the required fields when using Sentry.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+## Create Privacy Manifest in your Editor
+
+First, you need to create a privacy manifest file in your iOS project. This file should be named `PrivacyInfo.xcprivacy` and should be placed in the root of your Xcode project. Follow the [Privacy manifest files](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files#4284009) guide by Apple to create the file.
+
+```xml {filename:PrivacyInfo.xcprivacy}
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>NSPrivacyAccessedAPITypes</key>
```
Comment (kahest) [inline docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx:32]: `h`: so user defaults are not used?

File: docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Create Privacy Manifest in your Editor
+
+First, you need to create a privacy manifest file in your iOS project. This file should be named `PrivacyInfo.xcprivacy` and should be placed in the root of your Xcode project. Follow the [Privacy manifest files](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files#4284009) guide by Apple to create the file.
+
+```xml {filename:PrivacyInfo.xcprivacy}
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
```
Comment (kahest) [inline docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx:31]: `m`: it's not part of the required reason guidelines, but we should also list the types and usages of processed data (`NSPrivacyCollectedDataTypes`) here, similar to https://github.com/getsentry/sentry-cocoa/blob/aa45f3606be62823e6f0fd8f0d6a704b16867e17/Sources/Resources/PrivacyInfo.xcprivacy#L5

File: docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    </array>
+</dict>
+</plist>
+```
+
+After adding the privacy manifest file, don't forget to alter the file properties to `bundle resource` and `Copy to Output Directory` as `Copy if newer`
+
+## Troubleshooting
+
+- For more information about generating the privacy manifest using Xamarin, read the [Privacy manifests](https://devblogs.microsoft.com/dotnet/apple-privacy-manifest-support/) guide by Microsoft.
```
Comment (kahest) [inline docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx:59]: good idea to link to Microsoft's guide here 👍

**Review by kahest — APPROVED** [review]
thanks for adding this! left a few comments for completeness

also we should merge this after `2.1.0` has been released

File: docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+<Note>
+The information and steps in this guide are still being worked on and might change because of new tools or updated [Apple requirements](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files).
+</Note>
+
+The SDK will provide to your project a privacy manifest file during the build process only containing the required information used by the SDK. If you are providing your own privacyinfo.xcprivacy, you'll need to include the below rules on your privacyinfo.xcprivacy file. This guide will help you create the privacy manifest for your Xamarin applications.
```
Comment (vivianyentran) [inline docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx:17]:
```suggestion
During the build process, the SDK will provide your project a privacy manifest file containing only the required information used by the SDK. If you are providing your own privacy manifest, you'll need to include the below rules in your `privacyinfo.xcprivacy` file. This guide will help you create the privacy manifest for your Xamarin applications.
```

File: docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+## Before You Start
+
+Note that the application privacy manifest covers all data usages (via `NSPrivacyCollectedDataTypes`) and API usages (via `NSPrivacyAccessedAPITypes`) for the *whole application* and *all included libraries*. The examples below reflect the current requirements for the Sentry SDKs alone. If you are using other libraries that access privacy-relevant APIs, you need to include them in the privacy manifest as well.
+
+## Create Privacy Manifest in your Editor
```
Comment (vivianyentran) [inline docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx:23]:
```suggestion
## Create a Privacy Manifest
```

File: docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            <array>
+                <string>35F9.1</string>
+            </array>
+        </dict>
+    </array>
+</dict>
+</plist>
+```
+
+After adding the privacy manifest file, don't forget to alter the file properties to `bundle resource` and `Copy to Output Directory` as `Copy if newer`
```
Comment (vivianyentran) [inline docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx:55]:
```suggestion
After adding the privacy manifest file, don't forget to alter the file properties to `bundle resource` and `Copy to Output Directory` as `Copy if newer`.
```

**Review by vivianyentran — APPROVED** [review]
Added some small edits, but otherwise looks good! Thanks for adding this!

File: docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx (hunk 1 above)
Comment (lucas-zimerman) [reply docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx:32]: Oh, no, the SDK is not accessing the NSUserDefaults.

File: docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx (hunk 2 above)
Comment (lucas-zimerman) [reply docs/platforms/dotnet/guides/xamarin/data-management/apple-privacy-manifest.mdx:31]: Types added

**Review by kahest — APPROVED** [review]

**Pull request merged**